### PR TITLE
config: remove travis config, add circle config

### PR DESCRIPTION
**Pull request by phoebeschmidt**
Counterpart to https://github.com/contentful/contentful.js/pull/309

Switch to circle. Do not run browser tests that require Sauce Labs for now, because of trouble with account. Keep config so we can quickly re-enable sauce lab tests in the future.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
+version: 2.1
+jobs:
+  unit:
+    docker:
+      - image: circleci/node:8
+    steps:
+      - run: node --version
+      - run: npm --version
```
Comment (Khaledgarbaya) [inline .circleci/config.yml]: Do we need `node --version` and `npm --version` here ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      - run:
+          name: Build Success
+          when: on_success
+          command: ./node_modules/.bin/codecov
+  integration:
+    docker:
+      - image: circleci/node:8
+    steps:
+      - run: node --version
+      - run: npm --version
```
Comment (Khaledgarbaya) [inline .circleci/config.yml]: Same comment

File: .circleci/config.yml (hunk 1 above)
Comment (phoebeschmidt) [reply .circleci/config.yml]: good q. I took this from a ci setup we had in another proj (maybe cli?). Fine to remove it if you also dont think it's necessary

**Pull request merged**